src/Microsoft.SymbolStore/KeyGenerators/FileKeyGenerator.cs

```diff
@@ -53,6 +53,7 @@ private IEnumerable<KeyGenerator> GetGenerators()
                 yield return new MinidumpKeyGenerator(Tracer, _file);
                 yield return new ELFFileKeyGenerator(Tracer, _file);
                 yield return new PEFileKeyGenerator(Tracer, _file);
+                yield return new MachOFatHeaderKeyGenerator(Tracer, _file);
                 yield return new MachOFileKeyGenerator(Tracer, _file);
```

Comment:
Are MachO fat and regular file different such that the key generators won't overlap?

Reply:
On a second read, it looks like if it's a flat file, the FAT one would return null and the other one returns the specific stream. Is this correct?

Reply:
Yes, the two formats have different header signatures which causes one to fail (return null) and the other to succeed.